controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -831,7 +832,7 @@ func (pp *portPublisher) newPodRefAddress(endpointPort Port, endpointIP, podName
 	if err != nil {
 		return Address{}, PodID{}, fmt.Errorf("unable to fetch pod %v:%v", id, err)
 	}
-	ownerKind, ownerName := pp.k8sAPI.GetOwnerKindAndName(pod, false)
+	ownerKind, ownerName := pp.k8sAPI.GetOwnerKindAndName(context.Background(), pod, false)
```

Comment:
This feels like an important discussion to have, 

I had to short-circuit here, as `newPodRefAdress` is used in `add*`, `update*` etc methods, which are assigned directly to `ResourceEventHandlerFuncs` at https://github.com/linkerd/linkerd2/blob/f14b7cd88059628522668bbe38328c2d6e6699dc/controller/api/destination/watcher/endpoints_watcher.go#L153
and thus have fixed func signatures. 

As the input is passed from the client-go, we don't really have a way to get propagate context from top. Any ideas here?

_This same problem applies with `ip_watcher` too_

Reply:
I guess the only way would be by putting the context in the structs, although the godoc for `context` explicitly warns against that. But in this case I think it's ok, as long as it doesn't complicate things too much.

Reply:
Using `context.Background()` here seems fine to me.  The events from the informer don't pass along a context and even if they did, would it even make sense to use it?  e.g. would you want a cancellation or timeout of the informer event to timeout/cancel this?  I don't think so.

Reply:
Agree! @adleong